common/workunit/workunit.cpp

```diff
@@ -5416,6 +5416,34 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
             StringArray unknownAttributes;
             Owned<CQueryOrFilter> orFilter;
 
+            void updateWUState(IPropertyTreeIterator* wuIter)
+            {
+                Owned<IRemoteConnection> conWorkUnitAborts = querySDS().connect("/WorkUnitAborts/", myProcessSession(), 0, SDS_LOCK_TIMEOUT);
+                IPropertyTree* WUAbortPTree = conWorkUnitAborts->queryRoot();
+                if (!WUAbortPTree)
+                    return;
+
+                ForEach(*wuIter)
+                {
+                    IPropertyTree& wu = wuIter->query();
+                    WUState state = (WUState) getEnum(&wu, "@state", states);
+                    switch (state)
+                    {
+                    case WUStateRunning:
+                    case WUStateDebugPaused:
+                    case WUStateDebugRunning:
+                    case WUStateBlocked:
+                    case WUStateCompiling:
+                        if (WUAbortPTree->queryPropTree(wu.queryName()))
+                            setEnum(&wu, "@state", WUStateAborting, states);
```

Comment:
This is a bit dodgy, you're connecting on an unlocked connecting, then modifying state behind potentially locked workunits.
You should use the interface to query the state: workunit->getState() - that returns state and checks if aborting() without modifying the workunit.